sklearn/model_selection/_classification_threshold.py

```diff
@@ -611,31 +608,9 @@ def _fit_and_score_over_thresholds(
     else:  # prefit estimator, only a validation set is provided
         X_val, y_val, score_params_val = X, y, score_params
 
-    if curve_scorer is roc_curve or (
-        isinstance(curve_scorer, _BaseScorer) and curve_scorer._score_func is roc_curve
-    ):
-        fpr, tpr, potential_thresholds = curve_scorer(
-            classifier, X_val, y_val, **score_params_val
-        )
-        # For fpr=0/tpr=0, the threshold is set to `np.inf`. We need to remove it.
-        fpr, tpr, potential_thresholds = fpr[1:], tpr[1:], potential_thresholds[1:]
-        # thresholds are in decreasing order
-        return potential_thresholds[::-1], ((1 - fpr)[::-1], tpr[::-1])
-    elif curve_scorer is precision_recall_curve or (
-        isinstance(curve_scorer, _BaseScorer)
-        and curve_scorer._score_func is precision_recall_curve
-    ):
-        precision, recall, potential_thresholds = curve_scorer(
-            classifier, X_val, y_val, **score_params_val
-        )
-        # thresholds are in increasing order
-        # the last element of the precision and recall is not associated with any
-        # threshold and should be discarded
-        return potential_thresholds, (precision[:-1], recall[:-1])
-    else:
-        scores, potential_thresholds = curve_scorer(
-            classifier, X_val, y_val, **score_params_val
-        )
+    scores, potential_thresholds = curve_scorer(
+        classifier, X_val, y_val, **score_params_val
+    )
```

Comment:
Let's make `score` return ``potential_thresholds, scores`` to have the same order returned by each function